### DISCUSSION TITLE
Infinitely scrolling PageView with auto_route<title>

### DISCUSSION THREAD

**Discussion by smartman0307** [community/community#56536] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I want to implement an infinitely scrolling, lazy loading Tik-Tok like social feed using Flutter and [auto_route](https://pub.dev/packages/auto_route).

**Comment by silverspoon19931120** [comment]
AutoTabsRouter(
  routes: [
    BooksTab(),
    ProfileTab(),
    SettingsTab(),
  ],
  builder: (context, child, _) {
    final tabsRouter = AutoTabsRouter.of(context);
    return Scaffold(
      appBar: AppBar(
        title: Text(context.topRoute.name),
        leading: AutoLeadingButton(),
      ),
      body: PageView.builder(
        itemCount: tabsRouter.tabs.length,
        controller: tabsRouter.pageController,
        itemBuilder: (context, index) {
          final tab = tabsRouter.tabs[index];
          return tab.router;
        },
        onPageChanged: (index) => tabsRouter.setActiveIndex(index),
      ),
      bottomNavigationBar: BottomNavigationBar(
        currentIndex: tabsRouter.activeIndex,
        onTap: tabsRouter.setActiveIndex,
        items: [
          BottomNavigationBarItem(label: 'Books', ...),
          BottomNavigationBarItem(label: 'Profile', ...),
          BottomNavigationBarItem(label: 'Settings', ...),
        ],
      ),
    );
  },
);

**Comment by tuves** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
